### PR TITLE
Update readme and cleanup root dir.

### PR DESCRIPTION
**Pull request by bindi-nagda**
Merging this will close issue #78.

@abarret Feel free to add/ modify the README. I want it to be in a form that looks presentable if I were to share a link to this repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,56 @@
...
 
-## FAC Precondioner Stragegy: VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.cpp
+# Dependencies
 
-Contains routines for smoothening the error using box relaxation, computing residual, prolongation, restriction and solving on coarsest level.
+You will need to [build and install IBAMR](https://ibamr.github.io/linux) and all its required dependencies. We recommend compiling the optimized builds of the dependencies if you simply want to try out some simulations. Additionally, you should compile IBAMR with CMake since we provide a CMakeLists.txt to compile the source code in this repository. 
+
+The simulations generated can be viewed by installing [VISit](https://visit-dav.github.io/visit-website/) on your machine.
```
Comment (abarret) [inline README.md:9]:
```suggestion
The simulations generated can be viewed by installing [VisIt](https://visit-dav.github.io/visit-website/) on your machine.
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,56 @@
...
 
-## FAC Precondioner Stragegy: VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.cpp
+# Dependencies
 
-Contains routines for smoothening the error using box relaxation, computing residual, prolongation, restriction and solving on coarsest level.
+You will need to [build and install IBAMR](https://ibamr.github.io/linux) and all its required dependencies. We recommend compiling the optimized builds of the dependencies if you simply want to try out some simulations. Additionally, you should compile IBAMR with CMake since we provide a CMakeLists.txt to compile the source code in this repository.
```
Comment (abarret) [inline README.md:7]:
```suggestion
You will need to [build and install IBAMR](https://ibamr.github.io/building and all its required dependencies. This library uses CMake, and IBAMR must also be compiled with CMake. Unless you intend to actively develop the code, we suggest compiling optimized versions of the dependencies.

Note that IBAMR and its dependencies can be built automatically with `autoibamr`, and we are compatible with this method of building: https://github.com/ibamr/autoibamr.
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,56 @@
...
+
+# Compiling the code
+
+1. In the project’s root directory, create a build directory so that all object files and executables are stored separately from the source code.
+
+```
+mkdir build
+```
+
+2. Enter the build directory and run cmake to configure and generate build files for our `multiphaseLib` library, and link it to IBAMR. The command below uses the `O3` compiler optimization flag to enable level 3 optimizations.
```
Comment (abarret) [inline README.md:19]:
```suggestion
2. In the build directory, run `cmake` to configure and generate makefiles for the library and examples. For optimized builds, we suggest using the compilation flags `-O3 -march=native`. Note that the flag `-march=native` enables hardware-specific flags that are unsuitable for cross-platform builds.
```

File: README.md [hunk 4]
```diff
@@ -1,11 +1,56 @@
...
+```
+
+2. Enter the build directory and run cmake to configure and generate build files for our `multiphaseLib` library, and link it to IBAMR. The command below uses the `O3` compiler optimization flag to enable level 3 optimizations. 
+
+
+```
+cd build
+cmake /path/to/CMakeLists.txt \
+    -DIBAMR_ROOT=/path/to/ibamr/install/dir \
+    -DCMAKE_CXX_FLAGS="-O3"
```
Comment (abarret) [inline README.md:26]:
```suggestion
    -DCMAKE_CXX_FLAGS="-O3 -march=native"
```

File: README.md [hunk 5]
```diff
@@ -1,11 +1,56 @@
...
+```
+Note: If cmake fails, then make sure you're using the same compilers that were used for compiling IBAMR. You may provide a path to your compilers by passing flags for `-DCMAKE_CXX_COMPILER`, `-DCMAKE_C_COMPILER` and `-DCMAKE_Fortran_COMPILER`.
```
Comment (abarret) [inline README.md:28]: We do not require the same compilers. It should work as long as the compilers are ABI consistent.
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -1,11 +1,56 @@
...
+cd frm
+./frm input2d \
+    -solver_ksp_initial_guess_nonzero FALSE \
+    -solver_ksp_monitor_true_residual \
+    -solver_ksp_type fgmres \
+    -solver_ksp_max_it 20 \
+    -solver_ksp_rtol 1e-14 \
+```
+
+5. While the code is being executed, you may view the log files to monitor progress. After the run is complete, you may use VISit to plot the simulations results.
```
Comment (abarret) [inline README.md:48]:
```suggestion
5. While the code is being executed, you may view the log files to monitor progress. After the run is complete, you may use VisIt to plot the simulations results.
```

File: README.md [hunk 7]
```diff
@@ -1,11 +1,56 @@
...
+
+# Additional Support
+
+If you run into issues compiling or running this code, you may contact nagdabindi@gmail.com or barrett@math.utah.edu for additional support.
```
Comment (abarret) [inline README.md:52]: Do you want to be contacted by email? What about opening an issue or discussion?
```suggestion
If you run into issues compiling or running this code, you may open a disussion on GitHub for additional support. 
```

File: README.md [hunk 8]
```diff
@@ -1,11 +1,56 @@
...
-+ **apply.cpp**: Used to test that the operator is correct. This code applies the operator to user-defined network velocity, solvent velocity and pressure. The result is checked against the analytical solution to ensure the operator is 2nd order accurate.
-+ **solver.cpp**: This solves the coupled two-phase system using a krylov method (GMRES). The krylov method requires the operator to be passed to it. 
-+ **smooth.cpp**: This uses the stand-alone box relaxation scheme with gauss-seidel type iterations to solve the fluid equations
+This repository contains the source code to simulate multiphase flows on adaptively refined grids. The solver implemented here is second order accuracte in $L^1$, $L^2$, $L^\infty$ norms for all solution variables of the Newtonian and non-Newtonian models. Our experiments demonstrate the solver is stable provided $\Delta t$ satisfies the imposed CFL condition. The solver can accurately resolve sharp gradients in the solution and, with the multigrid preconditioner, its behavior is independent of grid spacing. This AMR solver offers a major cost savings benefits by using refined grids in areas of interest and using coarser grids elsewhere.
```
Comment (abarret) [inline README.md:3]:
```suggestion
This repository contains the source code to simulate multiphase flows on adaptively refined grids. The solver implemented here is second order accuracte in $L^1$, $L^2$, $L^\infty$ norms for all solution variables of the Newtonian and non-Newtonian models. Our experiments demonstrate the solver is stable provided $\Delta t$ satisfies the imposed CFL condition. The solver can accurately resolve sharp gradients in the solution and uses an efficient multigrid preconditioner to solve the resulting saddle point system. This AMR solver offers substantial cost savings by using refined grids in areas of interest and using coarser grids elsewhere.
```

File: README.md [hunk 9]
```diff
@@ -1,11 +1,56 @@
...
+```
+mkdir build
+```
+
+2. Enter the build directory and run cmake to configure and generate build files for our `multiphaseLib` library, and link it to IBAMR. The command below uses the `O3` compiler optimization flag to enable level 3 optimizations. 
+
+
+```
+cd build
+cmake /path/to/CMakeLists.txt \
```
Comment (abarret) [inline README.md:24]: We should make sure that the options are mostly copy and paste.
```suggestion
cmake .. \
```

**Review by abarret — COMMENTED** [review]
Please also put a reference to the arxiv paper when it is submitted.

**Comment by bindi-nagda** [comment]
> Please also put a reference to the arxiv paper when it is submitted.

Done!

File: README.md [hunk 10]
```diff
@@ -1,11 +1,59 @@
...
+For additional details on the numerical solver and simulation results, please refer to our paper: [Adaptive Mesh Refinement for Two-Phase Viscoelastic Fluid Mixture Models](https://arxiv.org/abs/2409.19974).
 
-Contains routines for smoothening the error using box relaxation, computing residual, prolongation, restriction and solving on coarsest level.
+# Dependencies
+
+You will need to [build and install IBAMR](https://ibamr.github.io/building) and all its required dependencies. This library uses CMake, and IBAMR must also be compiled with CMake. Unless you intend to actively develop the code, we suggest compiling optimized versions of the dependencies.
+
+Note that IBAMR and its dependencies can be built automatically with `autoibamr`, and we are compatible with this method of building: https://github.com/ibamr/autoibamr.
+
+The simulations generated can be viewed by installing [VISit](https://visit-dav.github.io/visit-website/) on your machine.
```
Comment (abarret) [inline README.md:13]:
```suggestion
The simulations generated can be viewed by installing [VisIt](https://visit-dav.github.io/visit-website/) on your machine.
```

**Review by abarret — APPROVED** [review]

**Pull request merged**